syft/frameworks/torch/mpc/falcon/falcon_helper.py

```diff
@@ -67,6 +82,47 @@ def xor(
             result = result.wrap()
         return result
 
+    @staticmethod
+    def determine_sign(
+        x: ReplicatedSharingTensor, beta: ReplicatedSharingTensor
+    ) -> ReplicatedSharingTensor:
+        """
+        return: x if beta = 0 | -x if beta = 1
```

Comment:
I am not sure I understand how this is done here, 
according to Line 2 in the protocol `beta` should be secret shared (over p, not 2 !!)

Reply:
So if the result is in `Z2` the algorithm won't work, since some of the `u` values need to be negative to take `c`=0 , which is effectively what the private compare is looking for